### DISCUSSION TITLE
Is there a way to see the list of available printers and select?

### DISCUSSION THREAD

**Discussion by DOEHOONLEE** [microsoft/MicrosoftEdge-Extensions#72] [Discussions]
Hi~!
I am trying to see the list of available printers and select one for users. I know this sounds abnormal since they are on different level, web and os.
Though, I saw a workaround for this with "[Native Messaging](https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/developer-guide/native-messaging?tabs=v3%2Cmacos)", but it is not working for some reason throwing "_Failed to start native messaging host_".

What I am trying to do is to

OS: Mac, Windows

* make a script file like below to find available printers

```bash
lpstat -p
```

* execute above script with edge extension and get stdout

OR

* perhaps, somehow get the list of available printers and select specific printer to use

Could this be possible?

Here is the reference [video I watched](https://www.youtube.com/watch?v=Odei9r2SB78&t=1074s). Thank you!!

**Comment by sivsouvamMSFT** [comment]
Hi @DOEHOONLEE, thank you for bringing this up. With reference to the video, you are willing to open the printer applications using the extension or the list of available printers? Also wanted to mention while using the printing option, you would find the list of available printers in your network.
